cloud/aws/beanstalk/monitors-beanstalk.tf

```diff
@@ -0,0 +1,136 @@
+
+	program_text = <<-EOF
+		from signalfx.detectors.not_reporting import not_reporting
+		signal = data('InstanceHealth', filter=filter('stat', 'mean') and filter('namespace', 'AWS/ElasticBeanstalk') and ${module.filter-tags.filter_custom})
+		not_reporting.detector(stream=signal, resource_identifier=['InstanceId'], duration='${var.heartbeat_timeframe}').publish('CRIT')
+	EOF
+
+	rule {
+		description           = "has not reported in ${var.heartbeat_timeframe}"
+		severity              = "Critical"
+		detect_label          = "CRIT"
+		disabled              = coalesce(var.heartbeat_disabled, var.detectors_disabled)
+		notifications         = coalescelist(var.heartbeat_notifications, var.notifications)
+		parameterized_subject = "[{{ruleSeverity}}]{{{detectorName}}} {{{ruleName}}} on {{{dimensions}}}"
+	}
+}
+
+resource "signalfx_detector" "health" {
+	name = "${join("", formatlist("[%s]", var.prefixes))}[${var.environment}] AWS beanstalk health"
+
+	program_text = <<-EOF
+		signal = data('EnvironmentHealth', filter=filter('namespace', 'AWS/ElasticBeanstalk') and filter('stat', 'upper') and ${module.filter-tags.filter_custom})${var.health_aggregation_function}.${var.health_transformation_function}(over='${var.health_transformation_window}')
+		detect(when(signal >= ${var.health_threshold_critical})).publish('CRIT')
+		detect(when(signal >= ${var.health_threshold_warning})).publish('WARN')
+	EOF
+
+	rule {
+		description           = "is too high >= ${var.health_threshold_critical}"
+		severity              = "Critical"
+		detect_label          = "CRIT"
+		disabled              = coalesce(var.health_disabled_critical, var.health_disabled, var.detectors_disabled)
+		notifications         = coalescelist(var.health_notifications_critical, var.health_notifications, var.notifications)
+		parameterized_subject = "[{{ruleSeverity}}]{{{detectorName}}} {{{ruleName}}} ({{inputs.signal.value}}) on {{{dimensions}}}"
+	}
+
+	rule {
+		description           = "is too high >= ${var.health_threshold_warning}"
+		severity              = "Warning"
+		detect_label          = "WARN"
+		disabled              = coalesce(var.health_disabled_warning, var.health_disabled, var.detectors_disabled)
+		notifications         = coalescelist(var.health_notifications_warning, var.health_notifications, var.notifications)
+		parameterized_subject = "[{{ruleSeverity}}]{{{detectorName}}} {{{ruleName}}} ({{inputs.signal.value}}) on {{{dimensions}}}"
+	}
+
+}
+
+resource "signalfx_detector" "latency_p90" {
+	name = "${join("", formatlist("[%s]", var.prefixes))}[${var.environment}] AWS beanstalk applicaion latency p90"
+
+	program_text = <<-EOF
+		signal = data('ApplicationLatencyP90', filter=filter('namespace', 'AWS/ElasticBeanstalk') and filter('stat', 'lower') and filter('InstanceId', '*') and ${module.filter-tags.filter_custom})${var.latency_p90_aggregation_function}.${var.latency_p90_transformation_function}(over='${var.latency_p90_transformation_window}')
```

Comment:
stat should be upper no?

Reply:
upper and lower are valid.  It should be more sensitive this way